src/metabase/driver/mysql.clj

```diff
@@ -65,7 +65,8 @@
       kdb/mysql
       ;; 0000-00-00 dates are valid in MySQL, but JDBC barfs when queries return them because java.sql.Date doesn't allow it.
       ;; Add a param to the end of the connection string that tells MySQL to convert 0000-00-00 dates to NULL when returning them.
-      (update :subname (u/rpartial str "?zeroDateTimeBehavior=convertToNull"))))
+      ;; Also add params to force UTF-8 encoding of results
+      (update :subname (u/rpartial str "?zeroDateTimeBehavior=convertToNull&useUnicode=true&characterEncoding=UTF8&characterSetResults=UTF8"))))
```

Comment:
one possible concern with this approach is that we'd be forcing everyone to use UTF-8 even if that's not what they actually want.  pretty soon we are going to have to start thinking about how to allow for arbitrary connection string parameters on drivers so that things like this are possible on an as-needed basis.